calliope/backend/backends.py

```diff
@@ -518,8 +518,10 @@ def add_parameter(
             default=default,
             use_inf_as_na=use_inf_as_na,
         )
-        if not parameter_values.shape and parameter_da.isnull().all():
-            parameter_da = parameter_da.astype(float)
+        if parameter_da.isnull().all():
+            self._delete_pyomo_list(parameter_name, "parameters")
+            if not parameter_values.shape:
```

Comment:
What does this check mean? Does it check that it is a np.array or xr.DataArray? Can this be checked explicitly?

Reply:
It checks if there are any dimensions on the array. If there aren't and the value is NaN then the dtype of the "array" (which is actually not an array, since it isn't indexed over anything) is explicitly changed to match the NaN dtype (i.e., float). This ensures we don't get issues elsewhere.

Not sure how to make it more explicit. Maybe 
```python
is_array = parameter_values.shape
if not is_array:
    # NaN-filled dimensionless arrays must have a non-"object" dtype to 
    # avoid issues elsewhere.
    parameter_da = parameter_da.astype(float)
```

Reply:
In which case would `parameter_da.shape` return `None`? I tested with numpy and xarray - if the array is empty, I get an empty tuple. Other examples, like `np.nan.shape`, throw an error.

Reply:
Also, `parameter_da` can only be an xarray DataArray, so it would never have reason to call e.g. `np.nan.shape`

Reply:
Thanks, now I got it!